src/server/page_server.cpp

```diff
@@ -188,9 +188,33 @@ page_server::connection_handler::receive_stop_log_prior_dispatch (tran_server_co
 void
 page_server::connection_handler::receive_oldest_active_mvccid (tran_server_conn_t::sequenced_payload &a_sp)
 {
-  /* TODO dummy function. will be modified corretly. soon  */
-  const MVCCID oldest_mvccid = *reinterpret_cast<const MVCCID *const> (a_sp.pull_payload().c_str());
-  er_log_debug (ARG_FILE_LINE, "receive_oldest_active_mvccid(): %d\n", oldest_mvccid);
+  assert (m_server_type == transaction_server_type::PASSIVE);
+
+  const auto oldest_mvccid = *reinterpret_cast<const MVCCID *const> (a_sp.pull_payload().c_str());
+  const auto channel_id = get_channel_id ();
+  std::lock_guard<std::mutex> lockg { m_ps.m_pts_oldest_active_mvccids_mtx };
+
+  /*
+   * 1. The entry is already created when ths PTS is connected.
+   * 2. It is updated by the PTS only when it move foward.
+   *    Without update, it is MVCCID_ALL_VISIBLE by default, which is lesser than any mvccid assigned.
```

Comment:
`MVCCID_ALL_VISIBLE` is actually a guard value.
At least, as far as I understood, that's how it will be used.

Reply:
Yes, it is. This comment explains why `assert (m_ps.m_pts_oldest_active_mvccids[channel_id] < oldest_mvccid);` is always true. I'll elaborate more on it in the comment when it is initialized (set_passive_tran_server_connection).